src/window_handle.rs

```diff
@@ -310,6 +310,9 @@ impl WindowHandle {
             }
         }
         if was_focused != cx.app_state.focus {
+            // What is this for? If you call ViewId.request_focus(), this
+            // causes focus to be set to your view, then briefly set to
+            // None here and then back.  Why?
```

Comment:
the app_state focus will taken from it's option before a pointer down even is propagated. Once event propagation is done, if another view has set itself to have focus because of the pointer down event the focus_changed function will notify the old was_focused that it's focus has been lost and the new app state focus that it has gained focus

Reply:
What I found (I added dumping the stack to the focus method it calls if the value was `None`) is that, when `someViewId.request_focus()` was called with an existing focus target, focus goes old_id -> None -> new_id.

I didn't check if events were fired for focus going to `None`, since event propagation wasn't working at that point.  As long as we're not doubling up event notifications, it's probably fine - just peculiar.